### PR TITLE
Define docker_collector_metrics_whitelist option

### PR DESCRIPTION
**Pull request by rclagett**
## Pull Request Checklist

**Is this in reference to an existing issue?**

#### General

- [x] Update Changelog following the conventions laid out on [Keep A Changelog](http://keepachangelog.com/)

- [ ] Update README with any necessary configuration snippets

- [x] RuboCop passes

- [ ] Existing tests pass

#### New Features

- [ ] Tests

- [ ] Add any required documentation to to the README

#### Purpose
Define docker_collector_metrics_whitelist

### CODE REVIEW

**Comment by rclagett** [comment]
This change should have been part of https://github.com/Netuitive/chef-netuitive/pull/61 but was forgotten about.

**Comment by ziggythehamster** [comment]
The change looks good to me, but is `metrics_whitelist = ` (i.e. blank) going to do the right thing in the template? You might need `<% unless @docker_collector_metrics_whitelist.to_s == "" %>` or something along those lines leaving the line out if it's blank/nil

**Comment by rclagett** [comment]
I see what you mean.

What about `<% unless @docker_collector_metrics_whitelist.empty? %>`? 

It would follow the convention used for tags: https://github.com/Netuitive/chef-netuitive/blob/master/templates/default/netuitive-agent.conf.erb#L64

**Comment by ziggythehamster** [comment]
You're right. I always forget that `.empty?` isn't an ActiveSupport feature but built into the language. That's the proper fix :)

**Comment by rclagett** [comment]
Updated!

**Review by ziggythehamster — APPROVED** [review]
Looks good to me

**Comment by rclagett** [comment]
@majormoses Would you mind providing a review as well when you get a chance? Sorry we weren't able to get it right the first time 😞

File: libraries/resource_configure.rb [hunk 1]
```diff
@@ -15,6 +15,7 @@ class NetuitiveCookbook::NetuitiveConfigureResource < Chef::Resource::LWRPBase
   attribute(:disk_space_collector_exclude_filters, kind_of: [String, Array], default: '^/boot, ^/mnt')
   attribute(:disk_usage_collector_metrics_whitelist, kind_of: String, default: '(?:^.*\.io$|^.*\.average_queue_length$|^.*\.await$|^.*\.iops$|^.*\.read_await$|^.*\.reads$|^.*\.util_percentage|^.*\.write_await$|^.*\.writes$)')
   attribute(:docker_collector_enabled, kind_of: [TrueClass, FalseClass], default: false)
+  attribute(:docker_collector_metrics_whitelist, kind_of: String, default: '')
```
Comment (majormoses) [inline libraries/resource_configure.rb]:
```suggestion
  attribute(:docker_collector_metrics_whitelist, kind_of: [String, nil], default: nil)
```

File: templates/default/netuitive-agent.conf.erb [hunk 2]
```diff
@@ -174,7 +174,9 @@ metrics_whitelist = (?:^.*\.rx_byte$|^.*\.rx_errors$|^.*\.rx_packets$|^.*\.tx_by
 
 [[NetuitiveDockerCollector]]
 enabled = <%= @docker_collector_enabled ? 'True' : 'False' %>
+<% unless @docker_collector_metrics_whitelist.empty? %>
```
Comment (majormoses) [inline templates/default/netuitive-agent.conf.erb]:
```suggestion
<% unless @docker_collector_metrics_whitelist.nil? %>
```

**Review by majormoses — COMMENTED** [review]
What about defaulting to `nil` and checking that rather than empty? That seems more intentional to me.

**Comment by ziggythehamster** [comment]
I think you're right about changing the default. That said, empty and nil are both illegal for the config file I think, so maybe the empty check is still appropriate? Is there a use case for setting the white list regex to an empty string?

**Comment by rclagett** [comment]
An empty whitelist is perfectly acceptable in terms of the config file allowing it and the resulting behavior. It's the same as having no filtering done at all.

**Comment by rclagett** [comment]
We could set the default to `.*` which is effectively the same thing as an empty string.

**Comment by rclagett** [comment]
It would also mean checking for nil or empty wouldn't be necessary...

I'll push the commits for you to take a look. I can revert if need be.

**Review by majormoses — APPROVED** [review]
LGTM

**Comment by majormoses** [comment]
released: https://supermarket.chef.io/cookbooks/netuitive/versions/0.22.1

**Pull request merged**